sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp

```diff
@@ -477,7 +485,9 @@ namespace Azure { namespace Core { namespace Http {
       // in the wire.
       // By not moving the connection back to the pool, it gets destroyed calling the connection
       // destructor to clean libcurl handle and close the connection.
-      if (IsEOF())
+      auto lastStatusCode = static_cast<typename std::underlying_type<Http::HttpStatusCode>::type>(
```

Comment:
We don't need to keep the last Statuscode in the Curl session.  Instead we can put this logic in MoveConnectionBackToPool.
That function can do a 
`curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);` before taking the lock.  Based on the condition it will add it back to the pool or dispose of it.

Keeps the logic for when to add a CURL handle to the pool in the same place for all future callers.